### ISSUE TITLE
Issue - Passed contentType will be ignored because HttpEntity sets content type

### ISSUE THREAD

**Issue by sallu1001** [loopj/android-async-http#735]
I am uploading an image using loopj. Here I am sending different parameters using a JSONObject. I am encoding the image in Base64 while uploading it to the server. here is my code :- 
```java
try {
			jsonObj.put("place_id", "761");
			jsonObj.put("status", "1");
			jsonObj.put("type", "Photo");
			jsonObj.put("filename", "menu.jpg");
			jsonObj.put("filetype", "image/jpeg");
			jsonObj.put("user_id", "1");
			jsonObj.put("auth_id", "1");
			jsonObj.put("userfile",Base64.encode(inputData, Base64.DEFAULT));

		} catch (JSONException e1) {
			// TODO Auto-generated catch block
			e1.printStackTrace();
		}

		try {
		entity = new StringEntity(jsonObj.toString());
		entity.setContentEncoding(new BasicHeader(HTTP.CONTENT_TYPE, "application/json"));
		} catch (UnsupportedEncodingException e1) {
			// TODO Auto-generated catch block
			e1.printStackTrace();
		}
```
and using post request like this :- 
```java
client.post(getApplicationContext(), "my_url, "application/json",new AsyncHttpResponseHandler() 
```

Still getting the error "Passed contentType will be ignored because HttpEntity sets content type". Can someone help me with that.

**Comment by 79144876** [comment]
HI.I have not use HttpEntity. and same log.
```
  Gson gson = new Gson();
    String    body = gson.toJson(mySubObj);
 requestParams.put("body", body);
  asyncHttpClient.post(JniApi.appcontext, url, createAuthHeader(url), requestParams, "application/json", responseHandler);
```

**Comment by smarek** [comment]
This situation should be resolved by fix applied for #838 
Also HttpEntity has priority in setting Content-Type, only if it doesn't provide the header, arguments passed one will be used
